src/foam/core/ContextAgentRunnable.java

```
@@ -32,6 +37,14 @@ public void run() {
         Thread.currentThread().setName(description_);
       }
       agent_.execute(x_);
+    } catch (RuntimeException e) {
```

can `catch (Throwable t)` if we want to catch all exceptions.